CONTRIBUTING.md

```diff
@@ -0,0 +1,128 @@
+# Contributing To SwaySwap
+
+Thanks for your interest in contributing to SwaySwap! This document outlines the process for installing dependencies and setting up SwaySwap for development, as well as some conventions on contributing to SwaySwap.
+
+If you run into any difficulties getting started, you can always ask questions on our [Discord](https://discord.gg/xfpK4Pe).
+
+## Setting up a development workspace
+
+See the [introduction](../introduction/index.md) section for instructions on installing and setting up the Sway toolchain. (This step is only necessary if contributing to smart contract development with Sway.)
+
+You will need to have `yarn`, or a similar package manager installed. For the purposes of this contributing, we will recommend what was used in the original development of the project.
+
+```sh
+yarn install
+```
+
+## Getting the repository
+
+1. Visit the [SwaySwap](https://github.com/FuelLabs/swayswap) repo and fork the project.
+2. Then clone your forked copy to your local machine and get to work.
+
+```sh
+git clone https://github.com/FuelLabs/swayswap
+cd swayswap
+```
+
+## Building and testing
+
+**Sway:** The following steps will run the Sway test suite and ensure that everything is set up correctly. Note that this portion pertains to the Sway language, if your intention is to contribute to SwaySwap's front-end only, proceed to the next step.
+
+First, open a new terminal and start `fuel-core` with:
+
+```sh
+fuel-core
+```
+
+Then open a second terminal, cd into the `sway` repo and run:
+
+```sh
+cargo run --bin test
+```
+
+After the test suite runs, you should see:
+
+```console
+Tests passed.
+_n_ tests run (0 skipped)
+```
```

Comment:
Do we actually need them to install and test `forc` from source?

Reply:
Fair point, just thought it wouldn't hurt to be sure, since it's possible someone could be working with Sway/Forc on this project?

Reply:
Fair point, just thought it wouldn't hurt to be sure, since it's possible someone could be working with Sway/Forc on this project?

Reply:
They could, but then they could just install `forc` with `cargo install forc` and not do a checkout of the `sway` repo.